website/docs/r/emr_cluster.html.markdown

```diff
@@ -176,6 +176,51 @@ resource "aws_emr_cluster" "example" {
 }
 ```
 
+### Multiple Node Master Instance Group
+
+Available in EMR version 5.23.0 and later, an EMR Cluster can be launched with three master nodes for high availability. Additional information about this functionality and its requirements can be found in the [EMR Management Guide](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-plan-ha.html).
+
+```hcl
+# This configuration is for illustrative purposes and highlights
+# only relevant configurations for working with this functionality.
+
+# Map public IP on launch must be enabled for the subnet
```

Comment:
According to the docs, you can use private subnets too. From https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-plan-ha-launch.html:

> You can launch an EMR cluster with multiple master nodes in both public and private VPC subnets

Same comment applies to line 347.

Is this requirement enforced by the API? Obviously, assigning a public IP on a private subnet isn't going to do anything particularly useful :)

BTW, thanks very much for such a quick turnaround on the issue I raised!

Reply:
Ah, I see it might only be a requirement if the cluster is in a public subnet. Do you have a working private VPC subnet multi-master setup?

Reply:
Not yet; given that the restriction is only mentioned in the comments though, and not enforced in the implementation I should be able to give this a spin, time permitting, and verify.

Reply:
Confirmed.  Your change works just fine on a private EMR cluster without the private subnets having `map_public_ip_on_launch = true` in the config.

Reply:
Updated, thanks, @mattburgess!